4_visualize/src/plot_results.R

```diff
@@ -0,0 +1,412 @@
+    summarize(mean_temp = mean(temperature, na.rm=TRUE))
+  
+  # Build plot  
+  doy_plot <- ggplot()+
+    geom_line(data=glm_plot, aes(x=doy, y=temperature, color='a_gcm_glm'), size=0.5, alpha=0.2, linetype='solid') +
+    geom_line(data=glm_mean, aes(x=doy, y=mean_temp, color="b_gcm_glm_mean"), size=0.5, alpha=1, linetype='solid') +
+    geom_point(data = glm_preds_ice, aes(x=doy, y=temperature, color='c_gcm_glm_ice'), size=0.1, alpha=0.05, pch=16) +
+    scale_color_manual(values = c("cornflowerblue","midnightblue", "maroon4"), labels = c('GLM preds','mean of GLM preds', 'GLM ice'), guide= guide_legend(override.aes = list(alpha = c(1,1,1), linetype = c('solid','solid','blank'), shape=c(NA,NA,16), size=c(0.5,0.5,0.5)))) +
+    facet_grid(depth ~ period, labeller=labeller(.cols=label_both)) +
+    labs(title= paste(sprintf("%s", unique(run_group$site_id)), sprintf('GCM: %s', unique(run_group$gcm)), sep='\n'),  y="Temperature (\u00b0C)") +
+    theme_bw() +
+    theme(
+      panel.border = element_blank(),
+      strip.text = element_text(size=14),
+      axis.text = element_text(size=12),
+      plot.title = element_text(size=16, face="bold", hjust=0.5),
+      axis.title.x = element_text(size=16),
+      axis.title.y = element_text(size=16),
+      legend.text = element_text(size=14),
+      legend.title = element_blank(),
+      legend.position = "top"
+    )
+  
+  # Save plot
+  outfile <- sprintf(outfile_template, unique(run_group$site_id), unique(run_group$gcm))
+  ggsave(filename=outfile, plot=doy_plot, dpi=300, width=10, height=6)
+  return(outfile)
+}
+
+#' @title Plot predicted surface, middle, and bottom temperatures in a 
+#' given lake, for all 6 GCMs
+#' @description For a single lake, all 6 GCMs and all 3 time periods, 
+#' plot surface, middle, and bottom temperature predictions in each year, 
+#' by DOY, as well as the average GLM prediction for each DOY in each 
+#' 20-year time period. Also plot whether or not the model predicted ice 
+#' on a given DOY for any GCM or for all 6 GCMs. The output plot is 
+#' faceted by period and by depth class
+#' @param lake_group a single group from the `p2_glm_uncalibrated_lake_groups`
+#' grouped version of the `p2_glm_uncalibrated_runs` output tibble subset 
+#' to the site_id, gcm, time_period, raw_meteo_fl, export_fl, and 
+#' export_fl_hash columns and grouped by site_id. Then filtered to 
+#' only groups for which glm_success==TRUE for all runs in that group. 
+#' The function maps over these groups.
+#' @param outfile_template The template filepath for the exported png, which
+#' is customized with the site_id
+#' @return The filepath of the exported png 
+plot_20yr_average_preds_ice <- function(lake_group, outfile_template) {
+  # Get glm preds for the current lake (wide format)
+  glm_preds <- get_site_preds(lake_group$raw_meteo_fl, lake_group$export_fl, lake_group$gcm)
+
+  # Munge them to long format
+  glm_preds_long <- munge_long(glm_preds)
```

Comment:
This happens in multiple plotting functions. Seems like we could do this as a target elsewhere, but then there would be some file i/o happening, so not sure which is slower.

Reply:
Moved to be a separate target in `4_visualize.R`